bpfd/src/bpf.rs

```diff
@@ -311,6 +317,18 @@ impl BpfManager {
                     map_pin_path.expect("map_pin_path must be set after successfult load"),
                 )
                 .await?;
+
+                // Update the local copy with the modified map_used_by
+                if let Some(prog) = self.programs.get_mut(&id) {
```

Comment:
Wait we shouldn't have to do a `self.programs.get_mut(&id)` here the `program` var is a pointer to the actual program we just loaded.